dev/src/write-batch.ts

```diff
@@ -603,23 +580,25 @@ export class WriteBatch implements firestore.WriteBatch {
    * this request.
    * @returns  A Promise that resolves when this batch completes.
    */
-  async _commit(commitOptions?: {
+  async _commit<Req, Resp>(commitOptions?: {
```

Comment:
This should be `Req extends api.ICommitRequest`, otherwise your code below isn't really typesafe and you are enforcing `api.ICommitRequest` only when you create the request. Let me know if that adds any type issues (I believe it didn't for me).

Reply:
Good catch, thanks!